src/main/webapp/teacher/controlGameInstance/controlGameInstance.js

```diff
@@ -273,6 +288,8 @@ function initQuestionStatsListener({ gameInstanceId, currentQuestionId } = {}) {
 }
 
 function updateCurrentQuestionStats(updatedQuestionStats) {
```

Comment:
Maybe you can do an if to verify that the change in the doc was a change in the stats? That way this function is not activated unless it's necessary

Reply:
Right well thought! I will store the "state" in some general variables to only update the necessary parts when we receive updates.

Reply:
Done!